src/tasks/payload/payload_task.c

```diff
@@ -135,7 +135,39 @@ void beacon_down_command_test(slate_t *slate)
     }
 }
 
-void ping_command_test(slate_t *slate)
+bool heartbeat_check(slate_t *slate)
+{
+    const uint64_t TIME_THRESHOLD = 1000;
+    if (slate->is_payload_on)
+    {
+        bool response_received = ping_command_test(slate);
+        uint64_t timeDelta = absolute_time_diff_us(
+            slate->payload_most_recent_ping_time, get_absolute_time());
+        if (timeDelta >= TIME_THRESHOLD)
+        {
+            payload_turn_off(slate);
```

Comment:
We may need to pivot on how we're restarting the Raspberry pi slightly. Context: https://ssi-teams.slack.com/archives/C2L29KW8J/p1747723117424069